tools/lkl/lib/virtio_net_tap.c

```diff
@@ -122,5 +196,24 @@ struct lkl_netdev *lkl_netdev_tap_create(const char *ifname)
 		return NULL;
 	}
 
+	#ifdef CONFIG_EVENTFD
```

Comment:
It is really hacky, but it was the best I could think of. The problem is
that if eventfd has been hijacked then we get an LKL fd out of the call and
we can't poll on it in the same fd set as the host fd attached to the TAP
device. And since there's no way to signal to eventfd that we want to hit
the host, we would need to dlsym in a pointer to the host version.

There's also no easy way to tell if we've been hijacked -- we could set
some variable in a hijack constructor function and check it in here, but
then the hijack logic is spread around the whole codebase. I figured we
would want to keep the hijack logic out of the main codebase because we
won't always want to use it.

We do depend heavily on host functions here, but TAP is POSIX only so it's
safe. Also hiding these calls behind the host_ops stuff doesn't prevent it
from being hijacked.

What do you think is the best thing to do? We could add another CONFIG
option to turn on this behavior.
On Mar 16, 2016 6:26 PM, "Yuan Liu" <notifications@github.com> wrote:

> In tools/lkl/lib/virtio_net_tap.c
> <https://github.com/lkl/linux/pull/130#discussion_r56443682>:
>
> > @@ -122,5 +196,24 @@ struct lkl_netdev *lkl_netdev_tap_create(const char *ifname)
> >  		return NULL;
> >  	}
> >
> > +	#ifdef CONFIG_EVENTFD
>
> Using CONFIG_EVENTFD seems very hacky. What if I want lkl_sys_eventfd and
> to close the virtio? We are calling all kinds of host sys calls in this
> file. Doesn't it look fragile?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lkl/linux/pull/130/files/768b8f7f19c4c8f6c3f9d71531e68ba9d4662a73#r56443682>
>